openmetadata-service/src/main/java/org/openmetadata/service/security/auth/BasicAuthenticator.java

```diff
@@ -0,0 +1,477 @@
+    if (storedUser.getIsBot() != null && storedUser.getIsBot()) {
+      throw new IllegalArgumentException("User are only allowed to login");
+    }
+    RefreshToken refreshToken = validateAndReturnNewRefresh(storedUser.getId(), request);
+    JWTAuthMechanism jwtAuthMechanism =
+        JWTTokenGenerator.getInstance()
+            .generateJWTToken(storedUser.getName(), storedUser.getEmail(), JWTTokenExpiry.OneHour, false);
+    JwtResponse response = new JwtResponse();
+    response.setTokenType("Bearer");
+    response.setAccessToken(jwtAuthMechanism.getJWTToken());
+    response.setRefreshToken(refreshToken.getToken().toString());
+    response.setExpiryDuration(jwtAuthMechanism.getJWTTokenExpiresAt());
+
+    return response;
+  }
+
+  public void verifyPasswordResetTokenExpiry(PasswordResetToken token) {
+    if (token.getExpiryDate().compareTo(Instant.now().toEpochMilli()) < 0) {
+      throw new RuntimeException(
+          "Password Reset Token" + token.getToken() + "Expired token. Please issue a new request");
+    }
+    if (Boolean.FALSE.equals(token.getIsActive())) {
+      throw new RuntimeException("Password Reset Token" + token.getToken() + "Token was marked inactive");
+    }
+  }
+
+  public RefreshToken validateAndReturnNewRefresh(UUID currentUserId, TokenRefreshRequest tokenRefreshRequest)
+      throws JsonProcessingException {
+    String requestRefreshToken = tokenRefreshRequest.getRefreshToken();
+    RefreshToken storedRefreshToken = (RefreshToken) tokenRepository.findByToken(requestRefreshToken);
+    if (storedRefreshToken == null) {
+      throw new RuntimeException("Invalid Refresh Token");
+    }
+    if (storedRefreshToken.getExpiryDate().compareTo(Instant.now().toEpochMilli()) < 0) {
+      throw new RuntimeException("Expired token. Please login again : " + storedRefreshToken.getToken().toString());
+    }
+    // TODO: currently allow single login from a place, later multiple login can be added
+    // just delete the existing token
+    tokenRepository.deleteTokenByUserAndType(currentUserId.toString(), REFRESH_TOKEN.toString());
+    // we use rotating refresh token , generate new token
+    RefreshToken newRefreshToken = TokenUtil.getRefreshToken(currentUserId, UUID.randomUUID());
+    // save Refresh Token in Database
+    tokenRepository.insertToken(newRefreshToken);
+    return newRefreshToken;
+  }
+
+  private User getUserFromRegistrationRequest(RegistrationRequest create) {
+    String username = create.getEmail().split("@")[0];
+    String hashedPwd = BCrypt.withDefaults().hashToString(12, create.getPassword().toCharArray());
+
+    BasicAuthMechanism newAuthMechanism = new BasicAuthMechanism().withPassword(hashedPwd);
+    return new User()
+        .withId(UUID.randomUUID())
+        .withName(username)
+        .withFullyQualifiedName(username)
+        .withEmail(create.getEmail())
+        .withDisplayName(create.getFirstName() + create.getLastName())
+        .withIsBot(false)
```

Comment:
isAdmin should be set to false as well. We should ignore or throw an error anyone coming from registration request as admin.
Only others admin can add an admin using the add user form.
Initial admins can only be added from openmetadata.yaml config